src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -411,6 +436,13 @@ class GroupReadsByUmi
       .filter(r => r.paired && r.mapped && r.mateMapped)
       .filter(r => r.refIndex == r.mateRefIndex)
       .filter(r => r.mapq >= this.minMapQ && r.get[Int](SAMTag.MQ.name()).exists(_ >= this.minMapQ))
+      .filter { r =>
+        this.minUmiLength.forall { l =>
```

Comment:
A few thoughts:
1) Perhaps we should add another filter here that just rejects reads that are missing UMIs?  Then everywhere else we can just use `rec[String](rawTag)`?
2) Post the other PR, this should probably count the rejected number of reads and log if > 0 were rejected
3) How about just counting non `-` instead?

Reply:
Yup. I was waiting on your PR being merged.